src/base/bittorrent/torrentimpl.cpp

```diff
@@ -1604,11 +1584,16 @@ void TorrentImpl::reload()
         p.flags &= ~(lt::torrent_flags::auto_managed | lt::torrent_flags::paused);
     }
 
+    const auto extensionData = new ExtensionData;
```

Comment:
This makes no sense, IMO, since right side clearly shows that it is a pointer.

Reply:
I really can't figure out (or even remember) when to use `auto *` or `auto` when facing pointers.
I would suggest **always use** `auto *` when the right hand side is a pointer, it makes it very explicit so the reader won't mix up with a variable.

Reply:
> I really can't figure out (or even remember) when to use `auto *` or `auto` when facing pointers.

You can always use `auto`. `auto *` is mostly for readers since right side doesn't always clearly shows that it is a pointer. `auto *` also fails if right side is mistakenly not a pointer.


> I would suggest **always use** `auto *` when the right hand side is a pointer, it makes it very explicit so the reader won't mix up with a variable.

Ok.